wordpress-mobile/WordPress-Android
# [Fix] Add targetSdk 34 foreground service types to our services.

Fixes #

The targetSdk 34 update now requires [foreground service types](https://developer.android.com/about/versions/14/behavior-changes-14#fgs-types). This PR adds it.

-----

## To Test:

We should 🚬 test all the services as a sanity check. There should be no changes. We can checklist them as they are tested.

Note: Use an Android 14 device.

   - [ ] .ui.reader.services.update.ReaderUpdateJobService
   - [ ] .ui.reader.services.discover.ReaderDiscoverJobService
   - [ ] .ui.reader.services.post.ReaderPostJobService
   - [ ] .ui.reader.services.search.ReaderSearchJobService
   - [ ] .ui.reader.services.comment.ReaderCommentService
   - [ ] .ui.suggestion.service.SuggestionService
   - [ ] .push.NotificationsProcessingService
   - [ ] .ui.notifications.services.NotificationsUpdateJobService
   - [ ] .util.analytics.service.InstallationReferrerJobService
   - [ ] .login.LoginWpcomService
   - [ ] .ui.sitecreation.services.SiteCreationService
   - [ ] .ui.stats.refresh.lists.widget.WidgetService
   - [ ] .push.GCMMessageService

If you want, simply use the app. 

* Login and out. A fresh install fires `InstallationReferrerJobService`.
* Go in the reader and make some comments and likes. 
* Change tabs in the reader to view different parts.
* Have a secondary account and make some likes and comments on your posts which will trigger the notification services above. 
* Get notifications. Dismiss some and click into some.
* Create a site.
* Set the widget and make sure the data is right.

This will pretty much cover everything you see above.

-----

## Regression Notes

1. Potential unintended areas of impact

    n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)

   - [x] .ui.reader.services.update.ReaderUpdateService
   - [x] .ui.reader.services.update.ReaderUpdateJobService
   - [x] .ui.reader.services.discover.ReaderDiscoverService
   - [x] .ui.reader.services.discover.ReaderDiscoverJobService
   - [x] .ui.reader.services.post.ReaderPostService
   - [x] .ui.reader.services.post.ReaderPostJobService
   - [x] .ui.reader.services.search.ReaderSearchService
   - [x] .ui.reader.services.search.ReaderSearchJobService
   - [x] .ui.reader.services.comment.ReaderCommentService
   - [x] .ui.suggestion.service.SuggestionService
   - [x] .push.NotificationsProcessingService
   - [x] .ui.notifications.services.NotificationsUpdateService
   - [x] .ui.notifications.services.NotificationsUpdateJobService
   - [x] .util.analytics.service.InstallationReferrerService
   - [x] .util.analytics.service.InstallationReferrerJobService
   - [x] .login.LoginWpcomService
   - [x] .ui.sitecreation.services.SiteCreationService
   - [x] .ui.stats.refresh.lists.widget.WidgetService
   - [x] .push.GCMMessageService

3. What automated tests I added (or what prevented me from doing so)

    n/a

-----

## PR Submission Checklist:
